### PR TITLE
Flexibility to provide backup engine on the helm chart

### LINKED ISSUE #5847: Flexibility to provide backup engine on the helm chart
#### Feature Description

Configure the database engine, while creating the Vitess cluster using the helm chart. 

#### Use Case(s)

Currently, the helm chart is supporting the built-in database backup engine. When I want to use the `xtrabackup` as a backup engine, there is no provision to pass the database engine. 

It would be great if we can able to pass the database engine as below. 
```
backup:
    enabled: false
    database_engine: xtrabackup
```

It would provide flexibility to configure the desired database backup engine.

### PR DESCRIPTION
**Pull request by evalsocket**
Fixed #5847

### CODE REVIEW

**Comment by evalsocket** [comment]
@deepthi I am reading the documentation for backup(https://vitess.io/docs/user-guides/backup-and-restore/). Should we provide all flag configurable via helm or just add backup_engine_implementation in the helm.

**Comment by deepthi** [comment]
> @deepthi @derekperkins I am reading the documentation for backup(https://vitess.io/docs/user-guides/backup-and-restore/). Should we provide all flag configurable via helm or just add backup_engine_implementation in the helm.

I think we should allow all xtrabackup flags to be configurable via helm.

**Comment by evalsocket** [comment]
@deepthi ready for first round of code review

**Comment by deepthi** [comment]
I'll defer to @derekperkins for the actual review and approval. Don't know much about helm.

**Comment by deepthi** [comment]
@derekperkins ping

File: helm/vitess/templates/_helpers.tpl [hunk 1]
```diff
@@ -152,7 +152,7 @@ done
 -restore_from_backup
     {{ end }}
 
--backup_storage_implementation=$VT_BACKUP_SERVICE
+  -backup_storage_implementation=$VT_BACKUP_SERVICE
```
Comment (derekperkins) [inline helm/vitess/templates/_helpers.tpl]: is this spacing correct?

**Comment by derekperkins** [comment]
Sorry for the delay. I had one question about indentation on the yaml. Also, if you could add an example to the readme, that'd be great. https://github.com/vitessio/vitess/tree/master/helm/vitess We'll go ahead and merge it after that.

File: helm/vitess/values.yaml [hunk 2]
```diff
@@ -71,35 +71,6 @@ config:
...
-    # # For the xtrabackup backup engine, required user that xtrabackup will use to connect to the database server
-    # xtrabackup_user: ""
-
-    # #For the xtrabackup backup engine, if greater than 0, use data striping across this many destination files to parallelize data transfer and decompression
-    # xtrabackup_stripes: 0
-
-    # # For the xtrabackup backup engine, size in bytes of each block that gets sent to a given stripe before rotating to the next stripe
-    # xtrabackup_stripe_block_size:
-
-
```
Comment (derekperkins) [inline helm/vitess/values.yaml]: I think that these should stay in the values.yaml file, we keep all/most of the options here.

**Comment by sougou** [comment]
What's the status on this PR?

**Comment by evalsocket** [comment]
It is ready for the review

**Comment by deepthi** [comment]
> It is ready for the review

@derekperkins ping

**Comment by deepthi** [comment]
Trying to re-run failed tests.

**Comment by deepthi** [comment]
hi @evalsocket can you confirm that the helm example works fine with these changes? Then I can approve and merge.

**Comment by sougou** [comment]
Now that we're deprecating helm #6439, I think we can close this PR. LMK if anyone has other thoughts.

**Comment by deepthi** [comment]
> Now that we're deprecating helm #6439, I think we can close this PR. LMK if anyone has other thoughts.

Closing now.

**Pull request closed without merging**